src/view/dataElements/xdmObject.jsx

```diff
@@ -57,101 +69,210 @@ const XdmObject = ({
+                  sandboxMeta={selectedSandboxMeta}
+                  schemaMeta={selectedSchemaMeta}
+                  previouslySavedSchemaInfo={
+                    initInfo.settings && initInfo.settings.schema
+                  }
+                />
+              )}
+            </div>
+          </div>
+        </div>
+      )}
     </div>
   );
 };
 
 XdmObject.propTypes = {
   initInfo: PropTypes.object.isRequired,
   formikProps: PropTypes.object.isRequired,
+  sandboxesMeta: PropTypes.array,
+  selectedSandboxMeta: PropTypes.object,
+  setSelectedSandboxMeta: PropTypes.func,
   schemasMeta: PropTypes.array,
+  schemasMetaStatus: PropTypes.string,
+  setSchemasMetaStatus: PropTypes.func,
   selectedSchemaMeta: PropTypes.object,
-  setSelectedSchemaMeta: PropTypes.func
+  setSelectedSchemaMeta: PropTypes.func,
+  schemaStatus: PropTypes.string
 };
 
 const XdmExtensionView = () => {
+  const [sandboxesMeta, setSandboxesMeta] = useState();
   const [schemasMeta, setSchemasMeta] = useState();
+  const [schemasMetaStatus, setSchemasMetaStatus] = useState();
+  const [selectedSandboxMeta, setSelectedSandboxMeta] = useState();
   const [selectedSchemaMeta, setSelectedSchemaMeta] = useState();
+  const [schemaStatus, setSchemaStatus] = useState();
 
   // TODO: break apart this extension view
   return (
     <ExtensionView
       getInitialValues={({ initInfo }) => {
-        return fetchSchemasMeta({
+        return fetchSandboxes({
           orgId: initInfo.company.orgId,
           imsAccess: initInfo.tokens.imsAccess
         })
-          .then(_schemasMeta => {
-            setSchemasMeta(_schemasMeta);
+          .then(_sandboxesMeta => {
+            let sandboxName = null;
+            // if sandboxes are returned from the api call
+            if (_sandboxesMeta.sandboxes && _sandboxesMeta.sandboxes.length) {
+              setSandboxesMeta(_sandboxesMeta);
+              // default to the first item in the list
+              let sandboxMeta = _sandboxesMeta.sandboxes[0];
+              // if a sandbox exists in settings and is in the list, choose that item
+              if (initInfo.settings && initInfo.settings.sandbox) {
+                sandboxMeta = _sandboxesMeta.sandboxes.find(
+                  _sandboxMeta =>
+                    _sandboxMeta.name === initInfo.settings.sandbox.name
+                );
+              }
+              if (sandboxMeta && sandboxMeta.name) {
+                sandboxName = sandboxMeta.name;
+              }
+              setSelectedSandboxMeta(sandboxMeta);
+            }
+
+            return fetchSchemasMeta({
+              orgId: initInfo.company.orgId,
+              imsAccess: initInfo.tokens.imsAccess,
+              sandboxName
+            });
+          })
+          .then(response => {
+            setSchemasMeta(response.results);
 
-            if (!_schemasMeta.length) {
+            if (!response.results.length) {
+              setSchemasMetaStatus(STATUS_ERROR || null);
```

Comment:
What's the `|| null` for? `STATUS_ERROR` is a constant with a string value, so it doesn't seem like `|| null` would ever come into play.

Reply:
I have the null to get around this error that WebStorm is telling me without `|| null`:

```
Argument type string is not assignable to parameter type ((prevState: undefined) => undefined) | undefined   Type string is not assignable to type (prevState: undefined) => undefined 
```

Can you suggest an alternative?

Reply:
Hmm...that looks like error that comes from TypeScript checking. You might want to make sure TypeScript checking is disabled for the project. When I remove `|| null` in WebStorm, I don't see any error.

Reply:
Okay, thanks. I removed the `|| null` instances